### PR TITLE
PlayState rename

### PR DESCRIPTION
**Pull request by tom-overton**
This PR renames all instances of `GlobalContext*` to `PlayState* play`. It's a rather large PR, but it was mostly just a find-and-replace. This also makes it so that in z_play itself, the GameState is referred to as `thisx`, and PlayState is referred to as `this`. This naming scheme is expected to eventually go away, as we will remove this/thisx from the codebase, but for now Kenix told me to go forward with this.

### CODE REVIEW

File: tools/namefixer.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
     "Actor_GetSwitchFlag": "Flags_GetSwitch",
     "atan_flip": "Math_Acot2F",
     "atans": "Math_Atan2S",
+    "globalCtx": "play",
+    "globalCtx2": "play2",
```
[the remark this answers is not in the document]
Comment (tom-overton) [reply tools/namefixer.py:26]: Sure, that makes sense.

**Review by hensldm — APPROVED** [review]
From what I've checked everything looks good.

File: src/overlays/actors/ovl_En_Fall/z_en_fall.c [hunk 2]
```diff
@@ -534,9 +532,9 @@ void EnFall_MoonsTear_Fall(EnFall* this, GlobalContext* globalCtx) {
     }
 }
 
-void EnFall_Update(Actor* thisx, GlobalContext* globalCtx) {
+void EnFall_Update(Actor* thisx, PlayState* play) {
     EnFall* this = THIS;
-    this->actionFunc(this, globalCtx);
+    this->actionFunc(this, play);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Fall/z_en_fall.c:537]:
```suggestion

    this->actionFunc(this, play);
```

**Review by EllipticEllipsis — APPROVED** [review]
All looks good in the files I checked. (Although now I have a todo list...)

**Comment by hensldm** [comment]
Tested the namefixer on a branch which worked mostly fine. Could you add `"Effect_GetGlobalCtx": "Effect_GetPlayState",` to the list of functions though.

File: tools/namefixer.py [hunk 3]
```diff
@@ -539,11 +540,20 @@
     "EffectSsKiraKira_SpawnSmall": "EffectSsKirakira_SpawnSmall",
     "EffectSsKiraKira_SpawnDispersed": "EffectSsKirakira_SpawnDispersed",
     "EffectSsKiraKira_SpawnFocused": "EffectSsKirakira_SpawnFocused",
+    "Effect_GetGlobalCtx": "Effect_GetPlayState"
```
Comment (hensldm) [inline tools/namefixer.py:543]:
```suggestion
    "Effect_GetGlobalCtx": "Effect_GetPlayState",
```

**Review by jpburnett — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Good work on this!

**Pull request merged**